src/handshake.rs

```diff
@@ -141,6 +141,10 @@ pub enum Error {
     UnexpectedHeader(String),
     /// The Sec-WebSocket-Accept header value did not match.
     InvalidSecWebSocketAccept,
+    /// The Host header domain is not allowed,
+    InvalidHost(String),
```

Comment:
How about just `Denied`? The host isn't really "invalid" (as in malformed), it's probably a very fine host. It's just not allowed in.

Reply:
I agree with the David here, `DeniedHost` is better because when I read `InvalidHost` I think of a host header with an invalid value.